.github/workflows/ci.yml

```diff
@@ -8,114 +8,72 @@ jobs:
-          restore-keys: |
-            ${{ runner.os }}-${{ hashFiles('yarn.lock') }}-
-            ${{ runner.os }}-
       - uses: actions/setup-node@v2
         with:
-          node-version: "16.13.x"
-      - name: Install package dependencies
+          cache: yarn
+          node-version-file: .nvmrc
+          registry-url: https://npm.pkg.github.com/
+      - run: yarn install --frozen-lockfile
         env:
-          PACKAGE_REGISTRY_ACCESS_TOKEN: ${{ secrets.PACKAGE_REGISTRY_ACCESS_TOKEN }}
-        run: |
-          npm config set "//npm.pkg.github.com/:_authToken" "$PACKAGE_REGISTRY_ACCESS_TOKEN"
-          yarn install --frozen-lockfile
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - run: yarn eslint
 
   prettier:
     if: github.ref != 'refs/heads/master'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        with:
-          path: "**/node_modules"
-          key: ${{ runner.os }}-${{ hashFiles('yarn.lock') }}-${{ hashFiles('**/package.json') }}
-          restore-keys: |
-            ${{ runner.os }}-${{ hashFiles('yarn.lock') }}-
-            ${{ runner.os }}-
       - uses: actions/setup-node@v2
         with:
-          node-version: "16.13.x"
-      - name: Install package dependencies
+          cache: yarn
+          node-version-file: .nvmrc
+          registry-url: https://npm.pkg.github.com/
+      - run: yarn install --frozen-lockfile
         env:
-          PACKAGE_REGISTRY_ACCESS_TOKEN: ${{ secrets.PACKAGE_REGISTRY_ACCESS_TOKEN }}
-        run: |
-          npm config set "//npm.pkg.github.com/:_authToken" "$PACKAGE_REGISTRY_ACCESS_TOKEN"
-          yarn install --frozen-lockfile
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - run: yarn prettier
 
   jest-react-16:
     if: github.ref != 'refs/heads/master'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        with:
-          path: "**/node_modules"
-          key: ${{ runner.os }}-${{ hashFiles('yarn.lock') }}-${{ hashFiles('**/package.json') }}
-          restore-keys: |
-            ${{ runner.os }}-${{ hashFiles('yarn.lock') }}-
-            ${{ runner.os }}-
       - uses: actions/setup-node@v2
         with:
-          node-version: "16.13.x"
-      - name: Install package dependencies
+          cache: yarn
+          node-version-file: .nvmrc
+          registry-url: https://npm.pkg.github.com/
+      - run: yarn install --frozen-lockfile
         env:
-          PACKAGE_REGISTRY_ACCESS_TOKEN: ${{ secrets.PACKAGE_REGISTRY_ACCESS_TOKEN }}
-        run: |
-          npm config set "//npm.pkg.github.com/:_authToken" "$PACKAGE_REGISTRY_ACCESS_TOKEN"
-          yarn install --frozen-lockfile
-      - name: Run unit tests
-        run: USE_REACT_17=false yarn test --ci
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: USE_REACT_17=false yarn test --ci
```

Comment:
Maybe a bit subjective, but I prefer that steps with a fairly readable one-line implementation (e.g. invoking a package script) don't include a name, since I find it easier to scan the output from the pipeline and understand the specific commands each step corresponds to.